web/src/pages/logs-page.tsx

```diff
@@ -109,14 +90,36 @@ const LogsPage: React.FC = () => {
     setFilters(updatedFilters);
   };
 
-  const handleTenantChange = (newTenant: string) => {
-    setTenantInURL(newTenant);
-    runQuery({ tenantToConsider: newTenant });
+  const updateQuery = (selectedFilters?: Filters, selectedTenant?: string): string => {
+    if ((!selectedFilters || Object.keys(selectedFilters).length === 0) && !selectedTenant) {
+      const defaultQuery = defaultQueryFromTenant();
+
+      setQueryInURL(defaultQuery);
+
+      return defaultQuery;
+    } else {
+      const updatedQuery = queryFromFilters({
+        existingQuery: query,
+        filters: selectedFilters,
+        attributes: availableAttributes,
+        tenant: selectedTenant,
+      });
+
+      setQueryInURL(updatedQuery);
+
+      return updatedQuery;
+    }
+  };
+
+  const handleRefreshClick = () => {
```

Comment:
Probably better to remove `handleRefreshClick` since it just calls `runQuery` and instead call `runQuery` directly.

Reply:
the signatures differ so TypeScript will complain about it.

Reply:
👍